### PR TITLE
Create GCLB in front of prod buckets

### PR DESCRIPTION
**Pull request by justinsb**
By the power of bash

### CODE REVIEW

File: k8s.gcr.io/ensure-prod-gclb.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ensure_billing "${PROJECT}"
+
+color 6 "Enabling the compute API: ${PROJECT}"
+enable_api "${PROJECT}" compute.googleapis.com
+
+color 6 "Creating Global Address"
+if ! gcloud compute addresses describe --global "${NAME}" >/dev/null 2>&1; then
+  gcloud compute addresses create "${NAME}" --description="IP Address for GCLB for binary artifacts" --global
+fi
+ip_addr=$(gcloud compute addresses describe prod-artifacts --global --format='value(address)')
```
Comment (cblecker) [inline k8s.gcr.io/ensure-prod-gclb.sh]: Shouldn't this use `${NAME}` too?

File: k8s.gcr.io/ensure-prod-gclb.sh [hunk 2]
```diff
@@ -0,0 +1,114 @@
+#!/bin/sh
```
Comment (cblecker) [inline k8s.gcr.io/ensure-prod-gclb.sh]:
```suggestion
#!/usr/bin/env bash
```

File: k8s.gcr.io/ensure-prod-gclb.sh [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+# limitations under the License.
+
+# This script creates & configures the GCLB in front of the GCS bucket.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_PATH=$(readlink -f "$0")
+SCRIPT_DIR=$(dirname "$SCRIPT_PATH")
```
Comment (cblecker) [inline k8s.gcr.io/ensure-prod-gclb.sh]: You can also now delete line 23. Unnecessary use of readlink.
```suggestion
SCRIPT_DIR="$(dirname "${BASH_SOURCE[0]}")"
```

File: k8s.gcr.io/ensure-prod-gclb.sh [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+. "${SCRIPT_DIR}/lib.sh"
+
+function usage() {
```
Comment (cblecker) [inline k8s.gcr.io/ensure-prod-gclb.sh]:
```suggestion
usage() {
```

File: k8s.gcr.io/ensure-prod-gclb.sh [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+fi
+
+color 6 "Reconciling GCLB SSL Certificate"
+if ! gcloud compute ssl-certificates describe "${NAME}" >/dev/null 2>&1; then
+  gcloud beta compute ssl-certificates create "${NAME}" --domains "${DOMAIN}"
+fi
+
+color 6 "Reconciling GCLB Target HTTPS Proxy"
+if ! gcloud compute target-https-proxies describe "${NAME}" >/dev/null 2>&1; then
+  gcloud compute target-https-proxies create "${NAME}" --url-map="${NAME}" --ssl-certificates=${NAME}
```
Comment (cblecker) [inline k8s.gcr.io/ensure-prod-gclb.sh]:
```suggestion
  gcloud compute target-https-proxies create "${NAME}" --url-map="${NAME}" --ssl-certificates="${NAME}"
```

File: k8s.gcr.io/ensure-prod-gclb.sh [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+else
+  gcloud compute target-https-proxies update "${NAME}" --url-map="${NAME}" --ssl-certificates=${NAME}
```
Comment (cblecker) [inline k8s.gcr.io/ensure-prod-gclb.sh]:
```suggestion
  gcloud compute target-https-proxies update "${NAME}" --url-map="${NAME}" --ssl-certificates="${NAME}"
```

File: k8s.gcr.io/ensure-prod-gclb.sh [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+color 6 "Reconciling Global Address"
+if ! gcloud compute addresses describe --global "${NAME}" >/dev/null 2>&1; then
+  gcloud compute addresses create "${NAME}" --description="IP Address for GCLB for binary artifacts" --global
+fi
+ip_addr=$(gcloud compute addresses describe prod-artifacts --global --format='value(address)')
+echo "Address: ${ip_addr}"
+
+color 6 "Reconciling GCLB Backend Bucket"
+if ! gcloud compute backend-buckets describe "${NAME}" >/dev/null 2>&1; then
+  gcloud compute backend-buckets create "${NAME}" --gcs-bucket-name=${BUCKET_NAME}
```
Comment (cblecker) [inline k8s.gcr.io/ensure-prod-gclb.sh]:
```suggestion
  gcloud compute backend-buckets create "${NAME}" --gcs-bucket-name="${BUCKET_NAME}"
```

File: k8s.gcr.io/ensure-prod-gclb.sh [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+else
+  gcloud compute backend-buckets update "${NAME}" --gcs-bucket-name=${BUCKET_NAME}
```
Comment (cblecker) [inline k8s.gcr.io/ensure-prod-gclb.sh]:
```suggestion
  gcloud compute backend-buckets update "${NAME}" --gcs-bucket-name="${BUCKET_NAME}"
```

File: k8s.gcr.io/ensure-prod-gclb.sh (hunk 2 above)
Comment (justinsb) [reply k8s.gcr.io/ensure-prod-gclb.sh]: Thanks - just made a similar change across the board in #233 - will extend that to use your `env` form

File: k8s.gcr.io/ensure-prod-gclb.sh (hunk 3 above)
Comment (justinsb) [reply k8s.gcr.io/ensure-prod-gclb.sh]: Hmmm... copied this one from @thockin's scripts...

File: k8s.gcr.io/ensure-prod-gclb.sh (hunk 1 above)
Comment (justinsb) [reply k8s.gcr.io/ensure-prod-gclb.sh]: Thanks - fixed!

File: k8s.gcr.io/ensure-prod-gclb.sh (hunk 4 above)
Comment (justinsb) [reply k8s.gcr.io/ensure-prod-gclb.sh]: You're right - apparently `function` is not considered correct.  Removed.

File: k8s.gcr.io/ensure-prod-gclb.sh (hunk 5 above)
Comment (justinsb) [reply k8s.gcr.io/ensure-prod-gclb.sh]: Thanks - fixed!

File: k8s.gcr.io/ensure-prod-gclb.sh (hunk 6 above)
Comment (justinsb) [reply k8s.gcr.io/ensure-prod-gclb.sh]: Fixed!

File: k8s.gcr.io/ensure-prod-gclb.sh (hunk 7 above)
Comment (justinsb) [reply k8s.gcr.io/ensure-prod-gclb.sh]: Done!

File: k8s.gcr.io/ensure-prod-gclb.sh (hunk 8 above)
Comment (justinsb) [reply k8s.gcr.io/ensure-prod-gclb.sh]: Done!

**Comment by justinsb** [comment]
Thanks for looking @cblecker .  All fixed up apart from the symlink discovery, which I copied and pasted from @thockin's scripts...

File: k8s.gcr.io/ensure-prod-gclb.sh (hunk 3 above)
Comment (cblecker) [reply k8s.gcr.io/ensure-prod-gclb.sh]: This is the standard format we use for k/k hack/ scripts:
https://github.com/kubernetes/kubernetes/blob/bf22bbdeadae97748e0fe46bfd925aa14f3e3acf/hack/verify-gofmt.sh#L23-L24

Comment (justinsb) [reply k8s.gcr.io/ensure-prod-gclb.sh]: Works for me :-)   I'll change this PR and send a separate PR to fix other deviations.

File: k8s.gcr.io/ensure-prod-gclb.sh [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+fi
+
+# The GCP project name.
+PROJECT="k8s-gcr-prod"
+
+# Name for cloud objects (url-map, gclb, etc)
+NAME=prod-artifacts
+
+# Name for the prod bucket
+BUCKET_NAME=prod-artifacts
```
Comment (thockin) [inline k8s.gcr.io/ensure-prod-gclb.sh]: whence this name?

File: k8s.gcr.io/ensure-prod-gclb.sh [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+    echo > /dev/stderr
+}
+
+if [ $# != 0 ]; then
+    usage
+    exit 1
+fi
+
+# The GCP project name.
+PROJECT="k8s-gcr-prod"
```
Comment (thockin) [inline k8s.gcr.io/ensure-prod-gclb.sh]: We'll probably need to rename this project, eh?

**Pull request merged**